gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/ModifiedDateRangeBasedFileFilter.java

```diff
@@ -0,0 +1,116 @@
+import org.apache.gobblin.configuration.ConfigurationKeys;
+
+
+/**
+ * A {@link CopyableFileFilter} that drops a {@link CopyableFile} if file modified time not between the loop back window
+ *  <code>sourceFs<code>
+ */
+@Slf4j
+public class ModifiedDateRangeBasedFileFilter implements CopyableFileFilter {
+
+  private final Properties props;
+  private Period minLookBackPeriod;
+  private Period maxLookBackPeriod;
+  private DateTime currentTime;
+  private DateTime minLookBackTime;
+  private DateTime maxLookBackTime;
+
+  public static final String CONFIGURATION_KEY_PREFIX = "gobblin.dataset.filter.";
+  public static final String MODIFIED_MIN_LOOK_BACK_TIME_KEY =
+      CONFIGURATION_KEY_PREFIX + "selection.modified.min.lookbackTime";
+  public static final String MODIFIED_MAX_LOOK_BACK_TIME_KEY =
+      CONFIGURATION_KEY_PREFIX + "selection.modified.max.lookbackTime";
+  public static final String DEFAULT_DATE_PATTERN_TIMEZONE = ConfigurationKeys.PST_TIMEZONE_NAME;
+  public static final String DATE_PATTERN_TIMEZONE_KEY = CONFIGURATION_KEY_PREFIX + "datetime.timezone";
+
+  public ModifiedDateRangeBasedFileFilter(Properties properties) {
+    this.props = properties;
+    PeriodFormatter periodFormatter =
+        new PeriodFormatterBuilder().appendDays().appendSuffix("d").appendHours().appendSuffix("h").toFormatter();
+    this.minLookBackPeriod = props.containsKey(MODIFIED_MIN_LOOK_BACK_TIME_KEY) ? periodFormatter.parsePeriod(
+        props.getProperty(MODIFIED_MIN_LOOK_BACK_TIME_KEY)) : new Period(DateTime.now().getMillis());
+    this.maxLookBackPeriod = props.containsKey(MODIFIED_MAX_LOOK_BACK_TIME_KEY) ? periodFormatter.parsePeriod(
+        props.getProperty(MODIFIED_MAX_LOOK_BACK_TIME_KEY)) : new Period(DateTime.now().minusDays(1).getMillis());
+    this.currentTime = properties.containsKey(DATE_PATTERN_TIMEZONE_KEY) ? DateTime.now(
+        DateTimeZone.forID(props.getProperty(DATE_PATTERN_TIMEZONE_KEY)))
+        : DateTime.now(DateTimeZone.forID(DEFAULT_DATE_PATTERN_TIMEZONE));
+    this.minLookBackTime = this.currentTime.minus(minLookBackPeriod);
+    this.maxLookBackTime = this.currentTime.minus(maxLookBackPeriod);
+  }
+
+  /**
+   * For every {@link CopyableFile} in <code>copyableFiles</code> checks if a {@link CopyableFile#getOrigin()#getPath()#getModificationTime()}
+   * + date between the min and max look back window on <code>sourceFs</code> {@inheritDoc}
+   *
+   * @see CopyableFileFilter#filter(FileSystem,
+   *      FileSystem, Collection)
+   */
+  @Override
+  public Collection<CopyableFile> filter(FileSystem sourceFs, FileSystem targetFs,
+      Collection<CopyableFile> copyableFiles) {
+    Iterator<CopyableFile> iterator = copyableFiles.iterator();
+
+    ImmutableList.Builder<CopyableFile> filtered = ImmutableList.builder();
+
+    while (iterator.hasNext()) {
+      CopyableFile file = iterator.next();
+      boolean fileWithInModWindow = isFileModifiedBtwLookBackPeriod(file.getOrigin().getModificationTime());
```

Comment:
Drop the assignment and move it inside the if condition in the next statement.

Reply:
Fixed